### ISSUE TITLE
Empêcher la creation consecutive de threads sur le forum

### ISSUE THREAD

**Issue by ArnaudCalmettes** [zestedesavoir/zds-site#3102]
Suite à une vague de flood cet après midi, je pense que ça serait pas mal qu'on ait un garde-fou pour empêcher la création consécutive de N threads sur le forum en imposant un délai (un quart d'heure ?) comme pour les posts. 

Cela éviterait de pourrir la home le temps que les modérateurs réagissent. En l'occurrence la home était ici remplie de spam en 5 minutes.

**Comment by GerardPaligot** [comment]
J'ai pas pu suivre le spam. C'était un bot ou un utilisateur ?

**Comment by ArnaudCalmettes** [comment]
Je saurais pas trop dire mais à vue de pif je dirais que c'était un type.

**Comment by Situphen** [comment]
On vient de se refaire spammer aujourd'hui ! Heureusement, le Staff a réagit très vite (- de 3 minutes) :D

**Comment by SpaceFox** [comment]
Du coup je tagge en bloquant. Ça mérite un hotfix.

**Comment by SpaceFox** [comment]
Hmmm non, ce n'est pas le même problème : apparemment aujourd'hui c'est pas des nouveaux topics mais des messages dans divers forums.

**Comment by SpaceFox** [comment]
On pourrait imaginer un système pour empêcher ça (vu qu'apparemment c'était manuel) : si tu postes 2 messages (identiques ?) à la suite en moins de X minutes, tu deviens suspect pendant Y temps et tu es limité à un message toutes les Z minutes. Et ton nouveau message est automatiquement accompagné d'une alerte.

**Comment by Nodraak** [comment]
Je pense que si deux messages consecutifs sont identiques on peut considerer ca comme du spam et bloquer simplement pendant X minutes. Un utilisateur legitime ne postera jamais deux fois le meme message.

**Comment by Situphen** [comment]
+1 Ça ne me semble pas trop compliqué à faire en plus !

**Comment by ArnaudCalmettes** [comment]
Je m'interroge sur le "identiques". En première itération ça me semble tout
à fait acceptable, mais il faudra voir si l'égalité stricte est _assez
efficace_ dans la durée (et ne pas trouver une correspondance plus floue +
pas trop coûteuse à calculer).

Dans tous les cas de mon point de vue ça doit venir "en sus" du garde-fou
sur la création successive de threads.

Arnaud

2015-10-20 12:20 GMT+02:00 Situphen notifications@github.com:

**Comment by SpaceFox** [comment]
Bien entendu.
Le 20 oct. 2015 12:45, "Arnaud Calmettes" notifications@github.com a
écrit :

**Comment by Eskimon** [comment]
Je suis pas sur qu'un tag bloquant soit justifié ici. En soit le site marche très bien sans, c'est juste chiant pour les modos qui vont devoir travailler si jamais un spammeur se pointe.

**Comment by GerardPaligot** [comment]
Je reprends notre caribou, est-ce que cette issue est réellement bloquante ? Est-ce qu'elle nécessite de bloquer la prochaine release ?

Si oui, il faudrait un dev de libre pour développer le fix décrit par @SpaceFox sur [son précédent message](https://github.com/zestedesavoir/zds-site/issues/3102#issuecomment-149510035). Cela ne devrait pas prendre trop de temps à faire. :)

**Comment by SpaceFox** [comment]
Ça n'a rien de bloquant en fait.

**Comment by nils-van-zuijlen** [comment]
Est-ce que ce problème n'aurait pas été résolu ? Actuellement on ne peut pas poster deux messages consécutifs (cad qu'on doit attendre 15 minutes entre deux messages).

**Comment by gllmc** [comment]
Non, on peut toujours créer des sujets consécutifs. Par contre, il y a un délai entre les posts d'un même thread.
